### DISCUSSION TITLE
Import CSS Files

### DISCUSSION THREAD

**Discussion by CMonjo** [vercel/next.js#12915] [Help] [answered]
Hey ! 

I'm creating this issue for asking about css importation. I have just started NextJS to develop a landing page. I don't want to write my CSS in styled-jsx and I was looking to use CSS files like in React.

I read that I could write my CSS with the css module (https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css) to have something like this :

`<p className={styles.error}>hey</p`

However with that I cannot do a class concatenation... That's why i want to use css files like react !

So... This is what i'm trying to do or it doesn't work :

```
import React from 'react';
import Head from "next/head";

export default function Newsletter() {
    return (
        <div>
            <Head>
                <link href="/static/styles.css" rel="stylesheet" />
            </Head>
            <div className={'container error'}>
                <p>NEWSLETTER BLOCK</p>
            </div>
        </div>
    );
}


```

And this is my css file : 

```
.container {
    width: 100%;
    height: 200px;
}

.error {
    background-color: red;
}
```

I've also try this but it doesn't work either :
https://github.com/zeit/next.js/issues/299#issuecomment-263146962

How can I get an item with two classes without using jsx-styled? Thank's a lot !

**Comment by Vadorequest** [comment]
I guess you better look at SASS support if you want to use non-css-in-js solutions. (but I'm not familiar with those, I prefer using css-in-js with Emotion)

**Reply by CMonjo** [reply]
Thank's for your answer, i've just tried SASS but that removes the use of global.css for templating :(

**Comment by sambhav-gore** [comment]
just import css, it works fine:

you can import it in the _app.js

something like this:
```
import "./YOUR_CSS.css";

... later
<div className="container"> Stuff </div>
```

and your css should have that class
```css
.container {
    padding: 20px;
}
```

however note that this approach is error prone because of collisions and harder to maintain, And Also has a performance impact as you would be loading all the CSS which is not needed on the rendered page. But you **want** to have all css in one file, it should work fine.

Sorry, before edit, I pasted the code solution for css module instead. I have updated the answer now.

**Reply by CMonjo** [reply]
Thank's! But it's what i've specify on my topic, i don't want this because à can't concat two classes :(

**Reply by sambhav-gore** [reply]
I have updated the answer, with className you can apply mutiple classes together. Glad that you got it solved though. cheers

**Comment by rafaelalmeidatk** [comment, marked as answer]
> `<p className={styles.error}>hey</p>`
> 
> However with that I cannot do a class concatenation... That's why i want to use css files like react !

You can! These values are just strings, you can concanate them:
```js
<p className={styles.error + ' ' + styles.foo}>hey</p>
// or with template strings
<p className={`${styles.error} ${styles.foo}`}>hey</p>
```

To make things easier you can use a package like [`classnames`](https://github.com/JedWatson/classnames):
```js
<p className={cx(styles.error, styles.foo)}>hey</p>
```

**Reply by CMonjo** [reply]
Hey !
Thank's for your answer :)

I've already try and i get this : 

Components/Newsletter/Newsletter.js :
```
import React from 'react';
import styles from './Newsletter.module.css';

export default function Newsletter() {
    return (
        <div>
            <div className={styles.test + styles.color}>
                <p>NEWSLETTER BLOCK</p>
            </div>
        </div>
    );
}
```
Components/Newsletter/Newsletter.module.css :
```
.test {
    background-color: green;
    padding: 40px;
}

.color {
    color: red;
}
```

On my website i get this
![result](https://i.imgur.com/q8C0DFF.png)
But the container has the class
![result](https://i.imgur.com/iPG1NVR.png)

Did u have an idea? Thank's again!!

**Reply by rafaelalmeidatk** [reply]
Notice that I added a space between the two class names, if you concat the two classes directly they will become just one: `'a' + 'b' = 'ab'`. If you use a helper function or the lib classnames you don't have to worry about this

**Reply by CMonjo** [reply]
Oh yes my bad
That's work! 

I will use classname then! Thank's a lot!
